src/observer.ts

```diff
@@ -52,6 +57,22 @@ export function observer<P extends object, TRef = {}>(
         memoComponent = memo(wrappedComponent)
     }
 
-    memoComponent.displayName = baseComponentName
+    memoComponent.displayName = `observer(${baseComponentName})`
+    copyStaticProperties(baseComponent as any, memoComponent as any)
+
     return memoComponent
 }
+
+function copyStaticProperties(base: React.FunctionComponent, target: React.FunctionComponent) {
+    // From all the potential static properties in React, propTypes and defaultProps are the only ones that work
+    // (modern) for function components
+    // One could wonder whether custom defined static fields should be copied over.
+    // However, at this moment it doesn't seem common to establish custom fields on function components.
```

Comment:
otherwise, we should bring in `react-hoist-non-statics` package, to make sure the right things get moved. But if feels kinda overkill, since the following isn't a pattern afaik:

```javascript
function Select() {

}

Select.Option = function() { // Some other component for example

}

export default observer(Select)
```

Reply:
It doesn't seem to be overkill, see #32